docs/kb/developer-resources/running-commands-on-the-host-os.md

```diff
@@ -0,0 +1,137 @@
++++
+date = "2016-08-01T00:00:00Z"
+lastmod = "2016-08-01T00:00:00Z"
+title = "Running Commands on the Host OS"
+weight = "999999"
+categories = [ "Knowledgebase", "Developer Resources" ]
++++
+
+The current version of Replicted is delivered as Docker images and run in containers on every supported operating system. The Replicated containers are built on top of a small Alpine Linux base image. Sometimes, it's desirable to access the host operating system and execute commands or change the environment in some way.
+
+{{< warning title="Warning" >}}
+Remember that Replicated supports many different operating systems that will be running a variety of kernels, package managers and more. Any script you choose to run on the host OS should be portable and run across any supported system.
+{{< /warning >}}
+
+Replicated can leverage the Docker socket file to create a container that can connect to the host over SSH and run commands over that SSH connection. Anything you execute on the host should be tested thoroughly across all operating systems. It is possible to make breaking changes to the host that cannot be recovered remotely.
+
+Finally, the below example uses a `curl | bash` command in a script. You should carefully consider any security and supportability concerns that come from changing the host itself.
+
+```yaml
+---
+replicated_api_version: "2.3.5"
+
+name: "Application"
+
+properties:
+  logo_url: http://www.replicated.com/images/logo.png
+  console_title: Application
+ 
+components:
+  - name: SSH
+    containers:
+      # Executes a bash script to generate keys
+      # Writes the public key to the host's authorized_keys
+      - source: public
+        image_name: debian
+        version: jessie
+        ephemeral: true
+        publish_events:
+          - name: SSH Key Created
+            trigger: container-stop
+            subscriptions:
+              - component: Bootstrap
+                container: debian
+                action: start
+        volumes:
+          - host_path: /root/.ssh
+            container_path: /.ssh
+            is_ephemeral: true
+          - host_path: /keys
+            container_path: /keys
+            is_ephemeral: true
+            
+        cmd: '["/create_ssh_key.sh"]'
+        config_files:
+          - filename: /create_ssh_key.sh
+            file_mode: 0700
+            contents: |
+              #!/bin/bash
```

Comment:
This runs in the debian container, which will have bash?  

But your point is valid below.